napari/layers/_image_layer/model.py

```diff
@@ -247,16 +259,28 @@ def interpolation_index(self, interpolation_index):
     def colormap(self):
         """string or ColorMap: Colormap to use for luminance images.
         """
-        return self._node.cmap
+        return self.colormap_name, self._node.cmap
 
     @colormap.setter
     def colormap(self, colormap):
-        try:
-            cmap = self._colormaps[colormap]
-        except KeyError:
-            pass
-
-        self._node.cmap = cmap
+        name = '[unnamed colormap]'
```

Comment:
Wow this got really complicated really fast. Please update the docstring accordingly :)

Reply:
Yep, will do followup PR. =)